core/src/main/java/org/apache/iceberg/rest/HTTPHeaders.java

```diff
@@ -0,0 +1,168 @@
+        .stream()
+        .collect(
+            Collectors.toMap(headers -> headers.get(0).name(), headers -> headers.get(0).value()));
+  }
+
+  /**
+   * Returns a {@link ListMultimap} representation of the headers. Header names are
+   * case-insensitive.
+   */
+  @Value.Lazy
+  default ListMultimap<String, String> asMultiMap() {
+    return entries().stream()
+        .collect(Collectors.groupingBy(h -> h.name().toLowerCase(Locale.ROOT)))
+        .values()
+        .stream()
+        .collect(
+            ImmutableListMultimap.flatteningToImmutableListMultimap(
+                headers -> headers.get(0).name(),
+                headers -> headers.stream().map(HTTPHeader::value)));
+  }
+
+  /** Returns all the entries in this group for the given name (case-insensitive). */
+  default List<HTTPHeader> entries(String name) {
+    return entries().stream()
+        .filter(header -> header.name().equalsIgnoreCase(name))
+        .collect(Collectors.toList());
+  }
+
+  /** Returns whether this group contains an entry with the given name (case-insensitive). */
+  default boolean contains(String name) {
+    return entries().stream().anyMatch(header -> header.name().equalsIgnoreCase(name));
+  }
+
+  /**
+   * Adds the given header to the current group if no entry with the same name is already present.
+   * Returns a new instance with the added header, or the current instance if the header is already
+   * present.
+   */
+  default HTTPHeaders addIfAbsent(HTTPHeader header) {
+    return contains(header.name())
+        ? this
+        : ImmutableHTTPHeaders.builder().from(this).addEntry(header).build();
+  }
+
+  /**
+   * Adds the given headers to the current group if no entries with same names are already present.
+   * Returns a new instance with the added headers, or the current instance if all headers are
+   * already present.
+   */
+  default HTTPHeaders addIfAbsent(HTTPHeaders headers) {
+    List<HTTPHeader> newHeaders =
+        headers.entries().stream().filter(e -> !contains(e.name())).collect(Collectors.toList());
+    return newHeaders.isEmpty()
+        ? this
+        : ImmutableHTTPHeaders.builder().from(this).addAllEntries(newHeaders).build();
+  }
+
+  static HTTPHeaders of(HTTPHeader... headers) {
+    return ImmutableHTTPHeaders.builder().addEntries(headers).build();
+  }
+
+  static HTTPHeaders fromMap(Map<String, ? extends Iterable<String>> headers) {
+    ImmutableHTTPHeaders.Builder builder = ImmutableHTTPHeaders.builder();
+    headers.forEach(
+        (name, values) -> values.forEach(value -> builder.addEntry(HTTPHeader.of(name, value))));
+    return builder.build();
+  }
+
+  static HTTPHeaders fromSimpleMap(Map<String, String> headers) {
+    ImmutableHTTPHeaders.Builder builder = ImmutableHTTPHeaders.builder();
+    headers.forEach((name, value) -> builder.addEntry(HTTPHeader.of(name, value)));
+    return builder.build();
+  }
+
+  static HTTPHeaders fromMultiMap(Multimap<String, String> headers) {
```

Comment:
same as I mentioned above. Do we actually have a use case for a bunch of these static methods? I would try to keep those to the amount we actually need

Reply:
If the methods are tested, is that a big deal? This one indeed has no usage today, but more auth managers in the future could use it.

Reply:
The Iceberg project typically tries to only add things to APIs that are needed for current use cases instead of planning for potential use cases that may or may not actually exist/happen (independent of whether there are tests for unused methods or not).
You also have to keep in mind that everything that's being added to `iceberg-core` can't be easily changed without going through an API deprecation cycle, so we're trying to be mindful about what's really required in terms of APIs

Reply:
Fair enough, let's remove all the static methods for now.